File: test_repo/player.cs
using System;

public class Player {
    public float Health = 100f;
    public float MaxHealth = 100f;
    public float X = 0f;
    public float Y = 0f;
    public float Speed = 5f;

    public void calculate_player_health(float damage) {
        Health -= damage;
        if (Health < 0) Health = 0;
        Console.WriteLine($"Player took {damage} damage. Current Health: {Health}");
    }

    public void Move(float horizontal, float vertical) {
        X += horizontal * Speed;
        Y += vertical * Speed;
        Console.WriteLine($"Player moved to ({X}, {Y})");
    }

    public void Heal(float amount) {
        Health += amount;
        if (Health > MaxHealth) Health = MaxHealth;
        Console.WriteLine($"Player healed {amount}. Current Health: {Health}");
    }
}

File: test_repo/player.py
class Player:
    def __init__(self):
        self.health = 100.0
        self.max_health = 100.0
        self.x = 0.0
        self.y = 0.0
        self.speed = 5.0

    def calculate_player_health(self, damage):
        """Calculates health after taking damage and handles death state."""
        self.health -= damage
        if self.health < 0:
            self.health = 0
        print(f"Python Player took {damage} damage. Health: {self.health}")

    def move(self, dx, dy):
        """Updates player position based on movement vector."""
        self.x += dx * self.speed
        self.y += dy * self.speed
        print(f"Python Player moved to ({self.x}, {self.y})")

    def reset_position(self):
        self.x = 0.0
        self.y = 0.0
